staging/kos/pkg/controllers/subnet/validator.go

```diff
@@ -0,0 +1,458 @@
+		// The data that affects validation changed, return all rivals so that
+		// they can be re-validated again as the conflict might have disappeared.
+		oldRivals = c.rivals
+		c.rivals = make([]k8stypes.NamespacedName, 0)
+	}
+	c.ownerSummary = s
+
+	return oldRivals
+}
+
+func doNotRetryList(e error) bool {
+	return k8serrors.IsUnauthorized(e) ||
+		k8serrors.IsBadRequest(e) ||
+		k8serrors.IsForbidden(e) ||
+		k8serrors.IsNotAcceptable(e) ||
+		k8serrors.IsUnsupportedMediaType(e) ||
+		k8serrors.IsMethodNotSupported(e)
+}
+
+func (v *Validator) recordConflicts(candidate *subnet.Summary, potentialRivals []netv1a1.Subnet) ([]string, bool, error) {
+	var conflictsMsgs []string
+	var conflictFound bool
+
+	for _, pr := range potentialRivals {
+		potentialRival, parsingErrs := subnet.NewSummary(&pr)
+		if len(parsingErrs) > 0 {
+			glog.Errorf("parsing %s failed while validating %s: %s", potentialRival.NamespacedName, candidate.NamespacedName, parsingErrs.Error())
+		}
+
+		if !potentialRival.Conflict(candidate) || potentialRival.SameSubnetAs(candidate) {
+			// potentialRival is not a rival to candidate or it is the same
+			// subnet as candidate, hence we skip it.
+			continue
+		}
+
+		// If we're here the two subnets represented by potentialRival and
+		// candidate are in conflict, that is, they are rivals.
+		conflictFound = true
+		if potentialRival.CIDRConflict(candidate) {
+			glog.V(2).Infof("CIDR conflict found between %s (%d, %d) and %s (%d, %d).", candidate.NamespacedName, candidate.BaseU, candidate.LastU, potentialRival.NamespacedName, potentialRival.BaseU, potentialRival.LastU)
+			conflictsMsgs = append(conflictsMsgs, fmt.Sprintf("CIDR overlaps with %s's (%s)", potentialRival.NamespacedName, pr.Spec.IPv4))
+		}
+		if potentialRival.NSConflict(candidate) {
+			glog.V(2).Infof("Namespace conflict found between %s and %s.", candidate.NamespacedName, potentialRival.NamespacedName)
+			conflictsMsgs = append(conflictsMsgs, fmt.Sprintf("same VNI but different namespace wrt %s", potentialRival.NamespacedName))
+		}
+
+		// Record the conflict in the conflicts cache.
+		if err := v.recordConflict(potentialRival, candidate); err != nil {
+			return conflictsMsgs, conflictFound, err
+		}
+	}
+
+	return conflictsMsgs, conflictFound, nil
+}
+
+func (v *Validator) updateSubnetValidity(s *netv1a1.Subnet, validated bool, validationErrors []string) error {
+	sCopy := s.DeepCopy()
+
+	sCopy.Status.Validated = validated
+	sCopy.Status.Errors.Validation = validationErrors
+
+	_, err := v.netIfc.Subnets(sCopy.Namespace).Update(sCopy)
+	switch {
+	case err == nil:
+		nsn := k8stypes.NamespacedName{
+			Namespace: s.Namespace,
+			Name:      s.Name,
+		}
+		v.updateStaleRV(nsn, s.ResourceVersion)
+	case doNotRetryUpdate(err):
+		glog.Errorf("failed to update subnet from %#+v to %#+v: %s. There will be no retry because of the nature of the error", s, sCopy, err.Error())
+	case !doNotRetryUpdate(err):
```

Comment:
s/case !doNotRetryUpdate(err)/default/

Reply:
Fixed in 8ac9883. But I'm not a big fan of default clause, it harms understandability of code. But this is pretty small stuff.